dart/lib/src/client.dart

```diff
@@ -77,6 +74,14 @@ abstract class SentryClient {
     return _options.transport.send(event);
   }
 
+  SentryEvent _prepareEvent(SentryEvent event) => event.copyWith(
```

Comment:
this method should do pretty much the same as `Scope.applyToEvent`, which is apply data if tis null on the event, but they have a different syntax, here is using `field: event.field ?? _options.field`

and on `Scope.applyToEvent`:

```
if (event.field == null) {
      event = event.copyWith(field: field);
    }
```

could we unify the syntax? either using the most friendly one for Dart? the result should be the same though, I believe the 2nd option would be easier to not allocate values if nothing to be copied from the Scope fields

Reply:
I prefer the first option, it is avoiding to instanciate a new Event for each field.

```dart
SentryEvent applyToEvent(SentryEvent event, dynamic hint) {
  return event.copyWith(
    transaction: event.transaction ?? transaction,
    user: event.user ?? user,
    // ...
}
```

Is it ok for you ?

Reply:
sounds good to me

Reply:
✅